### ISSUE TITLE
Expose execution ARN from aws_api_gateway_rest_api resource

### ISSUE THREAD

**Issue by mewa** [terraform-providers/terraform-provider-aws#3967]
Hi,

Currently, in order to create for example a Lambda permission for API Gateway one has to create an `aws_api_gateway_deployment` resource and obtain the `execution_arn` attribute.

```hcl
resource "aws_lambda_permission" "lambda_permission" {
  statement_id  = "AllowAPIGatewayInvoke"
  action        = "lambda:InvokeFunction"
  function_name = "example_lambda"
  principal     = "apigateway.amazonaws.com"

  source_arn = "${aws_api_gateway_deployment.example_deployment.execution_arn}/*/*"
}
```
However, this approach couples the deployment (stage) with specific permission.

Alternatively one could use a block like this
```hcl
  source_arn = "${replace(aws_api_gateway_deployment.example_deployment.execution_arn, format("/%s$/", aws_api_gateway_deployment.example_deployment.stage_name), "*")}/*/*"
```
to calculate the ARN not restricted to specific stage, or even create the ARN by hand according to the specification.

But either way we end up with messy expressions and/or permission coupled to the deployment.

I feel that the `aws_api_gateway_rest_api` resource should expose `execution_arn` in a fashion similar to `aws_api_gateway_deployment`.

**Comment by tdmalone** [comment]
I'd also love to see the `arn` exposed. Currently the `id` is exposed, but to build the `arn` you need to do something like:

    "arn:aws:apigateway:${data.aws_region.current.name}::/restapis/${aws_api_gateway_rest_api.api.id}"

**Comment by bflad** [comment]
Support for this has been merged in via #3968 and will release with v1.19.0 of the AWS provider, likely middle of next week.

@tdmalone yes please 😄  Should be a very similar PR to this one

**Comment by bflad** [comment]
This has been released in [version 1.19.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1190-may-16-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
